### PR TITLE
Fix email validation in API signup

### PR DESCRIPTION
**Pull request by jeffwang0516**
Found that user.full_clean() will trigger email validation, user.save() will not.
It raises ValidationError when the email is not valid.

### CODE REVIEW

File: smart_campus/app/models.py [hunk 1]
```diff
@@ -33,6 +33,7 @@ def create_user(self, email, password=None, nickname=""):
         )
 
         user.set_password(password)
+        user.full_clean()
         user.save(using=self._db)
```
Comment (Lee-W) [inline smart_campus/app/models.py]: Could you please briefly describe what `uesr.full_clean()` and `(using=self._db)` is for me?
Thanks in advance!

Comment (jeffwang0516) [reply smart_campus/app/models.py]: full_clean() will check if the data is valid for the fields in the model while save() will not, so it's possible for invalid data to enter the DB.

This issue if mainly for the EmailField in the User model.
Without full_clean(), invalid email will be directly saved.

Comment (jeffwang0516) [reply smart_campus/app/models.py]: Found that self._db is None, it can be removed

Comment (Lee-W) [reply smart_campus/app/models.py]: I see. Thanks!

**Comment by rapirent** [comment]
The ValidationError will not only raise cause email error input, but also will raise with max_length with nickname input.
I think we can just print the ValidationError's message, you can refer with [email validator](https://docs.djangoproject.com/en/1.11/ref/validators/#django.core.validators.EmailValidator.message) or [django source](https://docs.djangoproject.com/en/1.11/_modules/django/core/exceptions/#ValidationError)

File: smart_campus/app/models.py (hunk 1 above)
Comment (rapirent) [reply smart_campus/app/models.py]: The ValidationError will not only raise cause email error input, but also will raise with max_length like nickname input.
I think we can just print the ValidationError's message, you can check out with email validator or django source

Comment (jeffwang0516) [reply smart_campus/app/models.py]: Okay

**Pull request merged**